src/Core/Content/Product/SalesChannel/Review/ProductReviewSaveRoute.php

```diff
@@ -52,37 +51,30 @@ public function getDecorated(): AbstractProductReviewSaveRoute
     #[Route(path: '/store-api/product/{productId}/review', name: 'store-api.product-review.save', methods: ['POST'], defaults: ['_loginRequired' => true])]
     public function save(string $productId, RequestDataBag $data, SalesChannelContext $context): NoContentResponse
     {
-        $this->checkReviewsActive($context);
+        $salesChannelId = $context->getSalesChannelId();
+        if ($this->config->getBool('core.listing.showReview', $salesChannelId) === false) {
+            throw new ReviewNotActiveExeption();
+        }
 
-        /** @var CustomerEntity $customer */
         $customer = $context->getCustomer();
 
-        $languageId = $context->getContext()->getLanguageId();
-        $salesChannelId = $context->getSalesChannel()->getId();
+        // Should always be valid, as the route requires login
+        \assert($customer !== null);
 
-        $customerId = $customer->getId();
+        $data->set('customerId', $customer->getId());
+        $data->set('name', $customer->getFirstName());
+        $data->set('lastName', $customer->getLastName());
+        $data->set('email', $customer->getEmail());
```

Comment:
```suggestion
```
We can now remove this 3 here and directly set `'name' => $customer->getFirstName()` & `'email' => $customer->getEmail()` in the `$review` object. `lastName` is unsed and can be removed